### PR TITLE
SM-775 - Validate property length

### PR DESCRIPTION
**Pull request by rhvivancoeffio**
# Proposed Changes

### CODE REVIEW

File: internal/services/vehicle_signals_event_property_service.go [hunk 1]
```diff
@@ -45,6 +45,27 @@ func (v *vehicleSignalsEventPropertyService) GenerateData(ctx context.Context, d
 
 	for key, value := range eventAvailableProperties {
 
+		// Validate if value is required.
```
Comment (JamesReate) [inline internal/services/vehicle_signals_event_property_service.go:48]:
```suggestion
		// Validate if value is required to be considered for report
```

**Review by JamesReate — APPROVED** [review]
minor naming change

File: internal/services/vehicle_signals_event_property_service.go [hunk 2]
```diff
@@ -45,6 +45,27 @@ func (v *vehicleSignalsEventPropertyService) GenerateData(ctx context.Context, d
...
+		if _, ok := data[key]; ok {
```
Comment (JamesReate) [inline internal/services/vehicle_signals_event_property_service.go:49]: what if we put this in a function below eg. called `shouldValueBeConsidered` - if false, continue loop to skip current.

File: migrations/20240507143249_add_metadata_to_vehiclesignalsavailableproperties.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
+-- +goose Up
+-- +goose StatementBegin
+SELECT 'up SQL query';
+SET search_path = device_data_api, public;
+alter table vehicle_signals_available_properties add column min_length int;
```
[the remark this answers is not in the document]
Comment (rhvivancoeffio) [reply migrations/20240507143249_add_metadata_to_vehiclesignalsavailableproperties.sql:5]: The column was renamed

**Pull request merged**